About_equality_objects

daviddsr opened the pull request:
This is a koan to test equality methods in descendant classes of BasicObject and how this methods are overriden to provide specific class meaning.

alvarobp commented on src/about_equality_objects.rb:
```diff
@@ -0,0 +1,77 @@
+# This method is used by Hash to test members for equality neverminding order
+
+def test_equal_method_should_never_be_overriden_by_descendant_classes
+	assert_equal __(true), "string" == "string"
```
I think this is already covered in "test_equality_method_is_overriden_in_descendant_classes_to_provide_class_specific_meaning". I also think that the next two lines should be in that test. I think the only case that makes sense to show here is ``equal?``, since it looks like you want to show that we mustn't override equal?.

What do you think?

alvarobp commented:
I think you did a very good job, covering a very interesting (and useful) aspect of Ruby. This sets the ground for digging more on this topic.

Picky note: I think you should pay more attention to indentation. Every nested context should be indented and lines at the same context level should start at the same position.

alvarobp commented:
Another thing. Your commits belong to someone else. Make sure you configure git to identify you as yourself :P 

I'm not sure if we should help you fixing these commits or if you should reopen another Pull Request.

daviddsr commented:
I think you are right. I've changed the tests of test_equal_method_should_never_be_overriden_by_descendant_classes and I've put them in test_equality_method_is_overriden_in_descendant_classes_to_provide_class_specific_meaning, and made some changes as well to be more clear. The indentation problem I think is also fixed up.

alvarobp commented:
Cool!

One more thing about the indentation. This project uses a "2 spaces" indentation. I think you should use the same, otherwise the code is pretty inconsistent for a reader.

daviddsr commented:
The indentation is fixed up to two spaces instead tabs.

alvarobp commented:
:+1:

alvarobp commented:
Good job @daviddsr! 

Please configure git so commits belong to you. No one will know you made those commits otherwise.

daviddsr commented:
Right. I've already changed the git config propperly.

alvarobp commented:
@daviddsr Do you think it'd make sense to open a new Pull Request with the right commits? 

If you don't know how to do that I can help you doing that next Tuesday, since you need to create a new branch and change the author of the commits.

daviddsr commented:
ok, We'll do it next tuesday

esebastian commented:
Nice work! I'm closing the pull request so there's no confusion about the valid one, but it will be possible to continue the conversation with new comments when they arrive
